eo-maven-plugin/src/main/java/org/eolang/maven/tojos/TranspiledTojos.java

```diff
@@ -80,7 +86,9 @@ public void close() throws IOException {
      * @param second Xmir2 file.
      */
     public void add(final Path transpiled, final Path second) {
-        this.all.value().add(String.valueOf(transpiled)).set(Attribute.XMIR2.key(), second);
+        synchronized (this.lock) {
```

Comment:
@volodya-lombrozo why do you need explicit lock object here? Why not just
```
synchonized (this) {
}
```

Reply:
@mximp 

Lock operations on `this` may have unforeseen side-effects 
There are several reasons synchronization on `this` or class expressions may be a bad idea:
1. it makes synchronization a part of the external interface of the class, which makes a future change to a different locking mechanism difficult,
2. it becomes hard to track just who is locking on a given object,
3. it makes a denial-of-service attack possible, either on purpose or it can happen easily by accident when subclassing. 

As an alternative, consider synchronizing on a private final lock object, access to which can be completely controlled.